src/acl.js

```diff
@@ -0,0 +1,21 @@
+
+const isFunction = require('./utils/is-function')
+
+class ACL {
+  constructor (checkPermission) {
+    if (!isFunction(checkPermission)) {
+      throw new Error('Permission verification function is invalid')
+    }
+
+    this._checkPermission = checkPermission
+  }
+
+  async canAppend(key, entry) {
+    if (!key) throw new Error("A key is required to check for permission")
+    if (!entry.sig) throw new Error("Entry doesn't have a signature")
+    if (!entry.key) throw new Error("Entry doesn't have a public key")
```

Comment:
We should not check the shape/validity of the *Entry* in the AccessController, this is not its responsibility (but rather should be done in Entry).

Further, I believe we need the full `Identity` passed here in order to check whether a user is allowed to append to the log as we may (should, really) verify the identity as well.

Third, the default AccessController for the Log can be as simple as follows. The default access mode should be "the log can be appended to" and if this should not be the case for the user, then they should provide their version of the AccessController.
```javascript
async canAppend (identity, entry) {
  return true
}
```

Reply:
We can add the verifyIdentity function to provider, I agree that the identity should be verified. That means all entries need an identity and should include the chain signature.